MekHQ/src/mekhq/campaign/personnel/Person.java

```diff
@@ -516,44 +507,75 @@ public void setDependent(boolean tf) {
         }
     }
 
-    public boolean isPrisoner() {
-        return prisonerStatus == PRISONER_YES;
-    }
-
-    public void setPrisoner() {
-        prisonerStatus = PRISONER_YES;
-        setRankNumeric(Ranks.RANK_PRISONER);
-    }
-
-    public boolean isBondsman() {
-        return prisonerStatus == PRISONER_BONDSMAN;
-    }
-
-    public void setBondsman() {
-        prisonerStatus = PRISONER_BONDSMAN;
-        willingToDefect = false;
-        setRankNumeric(Ranks.RANK_BONDSMAN);
+    public PrisonerStatus getPrisonerStatus() {
+        return prisonerStatus;
     }
 
-    public boolean isFree() {
-        return (!isPrisoner() && !isBondsman());
+    public void setPrisonerStatus(PrisonerStatus prisonerStatus) {
+        setPrisonerStatus(prisonerStatus, true);
     }
 
-    public void setFreeMan() {
-        prisonerStatus = PRISONER_NOT;
-        willingToDefect = false;
-    }
+    public void setPrisonerStatus(PrisonerStatus prisonerStatus, boolean log) {
+        boolean freed = !this.prisonerStatus.isFree();
+        this.prisonerStatus = prisonerStatus;
 
-    public int getPrisonerStatus() {
-        return prisonerStatus;
-    }
+        // Now, we need to fix values and ranks based on the Person's status
+        switch (prisonerStatus) {
+            case PRISONER:
+            case PRISONER_DEFECTOR:
+                // They don't get to have a rank. Their rank is Bondsman.
+                getCampaign().changeRank(this, Ranks.RANK_BONDSMAN, true);
```

Comment:
Seems almost identical to the Bondsman functionality. Is this a) intended, and b) if it is, can it be consolidated?

Reply:
It is intended, albeit with a typo on the line in question (it should say RANK_PRISONER). However, I consolidated it. I'm also going to create a story to remove the RANK_BONDSMAN and RANK_PRISONER ranks and replacement with leading displays for ranks, as that is significantly more realistic.